src/datahelper.py

```diff
@@ -1,200 +1,404 @@
+                f"processing. \n At least {len(merge_metadata_data.items())-number_of_datapackages} csv files "
+                f"cannot be matched with metadata. Csv and metadata filenames must be identical! \n Ensure the "
+                f"parameter `number_of_datapackages` reflects the number of datapackages you want to upload."
+            )
+
+        for data_item in tqdm(
+            merge_metadata_data.values(),
+            desc="Fill resource column names in metadata with "
+            "postgresql-conform table headers from respective csv file",
+        ):
+
+            csv_column_header = data_item[0].columns
+
+            metadata_user = data_item[1]
+
+            # omi
+            # metadata 1.5 instance and parse metadata_user in python dict
+            dialect1_5 = OEP_V_1_5_Dialect()
+            # parse metadata_user in python dict
+            parsed: OEPMetadata = dialect1_5._parser().parse(  # pylint: disable=W0212
+                metadata_user
+            )
+
+            # similarity isn't case-agnostic. field.name.lower() -> to enable string comparison on lowercase
+            for ressource in parsed.resources:
+                for field in ressource.schema.fields:
+                    try:
+                        field.name = self.similar(csv_column_header, field.name.lower())
+                    except Exception as exc:
+                        raise Exception(
+                            f"There is a problem in metadata file: {metadata_user['name']}. "
+                            f"The metadata key `name` is: {field.name}"
+                        ) from exc
+
+            metadata = dialect1_5.compile_and_render(parsed)
+            metadata = json.loads(metadata)
+
+            self.write_json(
+                path=f"{self.output_dir}/{data_item[1]['name']}.json", file=metadata
+            )
+
+    def combine_dict(self, dict_1: dict, dict_2: dict) -> dict:
         """
-        The method inserts each csv specific version dicts in respective csvs.
-        :type columns: object
-        :param columns: Specify one of: version, other, all
-        :return:
+        This method merges two dicts even if the keys in the two dictionaries are different.
+        :param dict_1, dict_2: Two input dicts
+        :return: Merged dict
         """
-        version_dict = self.create_version_dict()
-        # todo: rewrite function that it takes arguments 1. version, 2. other - to have only one function that inserts version_json or json_dict_user_col in the other columns
-        for index, (filename, df) in enumerate(self.df_dict.items()):
-            # get matching-file-version_dict and insert in version column
-            # todo: integrate check whether each row is already filled with version or not
-            df["version"] = f"{version_dict.get(filename)}"
+        return {
+            k: tuple(d[k] for d in (dict_1, dict_2) if k in d)
```

Comment:
This can lead to different tuple lenghts!
As you expliciily use index "1" later (can't fd i righ now), you should make sure that tuple has always lengh of 2, like:
```
{
    k: tuple(d.get(k) for d in (dict_1, dict_2))
    for k in set(dict_1.keys()) | set(dict_2.keys())
}
```
as `get()` returns None if key is not found.